src/core/util/schemasUtil.ts

```diff
@@ -0,0 +1,253 @@
+      if (result) {
+        return result;
+      }
+    }
+    // some mappings are 'reversed'
+    if (typeof key === 'object') {
+      const result = doFindByUUID(key, uuid);
+      if (result) {
+        return result;
+      }
+    }
+  }
+  return undefined;
+};
+
+export const calculatePath = (
+  root: any,
+  object: any
+): Array<string> | CalculatePathError => {
+  const path = doCalculatePath(root, object);
+  if (!path) {
+    return {
+      id: 'calulatePathError',
+      root: root,
+      element: object,
+    };
+  }
+  return path;
+};
+
+export const getPathString = (object: any): string | PathError => {
+  const root = getRoot(object);
+  const path = calculatePath(root, object);
+  if (isPathError(path)) {
+    return path;
+  }
+  return `${path.join('/')}`;
+};
+
+const doCalculatePath = (root: any, object: any): Array<string> | undefined => {
+  if (root === object) {
+    return [];
+  }
+  const entries = root instanceof Map ? root.entries() : Object.entries(root);
+  for (const [key, value] of Array.from(entries)) {
+    if (value === object) {
+      return [key];
+    }
+    // some mappings are 'reversed'
+    if (key === object) {
+      return [value];
+    }
+    if (typeof value === 'object' && key !== 'parent') {
+      const path = doCalculatePath(value, object);
+      if (path) {
+        return [key, ...path];
+      }
+    }
+    // some mappings are 'reversed'
+    if (typeof key === 'object') {
+      const path = doCalculatePath(key, object);
+      if (path) {
+        return [value, ...path];
+      }
+    }
+  }
+  return undefined;
+};
+
+export const getFromPath = (
+  root: any,
+  path: Array<string>
+): any | GetPathError => {
+  const element = doGetFromPath(root, path);
+  if (!element) {
+    return {
+      id: 'getPathError',
+      root: root,
+      path: path,
+    };
+  }
+  return element;
+};
+
+const doGetFromPath = (root: any, path: Array<string>): any => {
+  if (path.length === 0) {
+    return root;
+  }
+  const [pathElement, ...rest] = path;
+  if (root instanceof Map) {
+    if (root.has(pathElement)) {
+      return getFromPath(root.get(pathElement), rest);
+    }
+    // must be a reverse map
+    const element = Array.from(root.entries()).reduce((acc, [key, value]) => {
+      if (value === pathElement) {
+        return key;
+      }
+      return acc;
+    }, undefined);
+    return getFromPath(element, rest);
+  }
+  return getFromPath(get(root, [pathElement]), rest);
+};
+
+export const linkElements = (
+  uiSchemaElement: EditorUISchemaElement,
+  schemaElement: SchemaElement
+): boolean => {
+  if (!uiSchemaElement.uuid) {
+    console.error('Found element without UUID', uiSchemaElement);
+    return false;
+  }
+
+  (schemaElement.linkedUISchemaElements =
+    schemaElement.linkedUISchemaElements || new Set()).add(
+    uiSchemaElement.uuid
+  );
+
+  uiSchemaElement.linkedSchemaElement = schemaElement.uuid;
+  return true;
+};
+
+export const linkSchemas = (
```

Comment:
I would like to suggest that after this method is executed schema and uischema should be in a consistent state in regards of linking, no matter how they did look before.

So both schemas need to be traversed completely, cleaned from old links and new ones set.

At the moment there are multiple issues:
 - If a new `undefined` or `false` schema is set, the uiSchema is not processed at all
 - When a new schema is set, only the links of controls which can resolve are set. However the new schema might not fit the old uischema. When that's the case the broken controls still point to non-existent uuids. These links should be removed.
 - The handling of the schema in case of new uischemas is handled in a different method (cleanLinked in the reducer). I would prefer it also be handled here so we have everything in place.